src/main/java/core/basesyntax/services/operations/PurchaseOperation.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.services.operations;
+
+import core.basesyntax.models.FruitTransaction;
+import core.basesyntax.services.OperationHandler;
+import core.basesyntax.services.StorageService;
+
+public class PurchaseOperation implements OperationHandler {
+    private StorageService storageService;
+
+    public PurchaseOperation(StorageService storageService) {
+        this.storageService = storageService;
+    }
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        String fruit = transaction.getFruit();
+        int quantity = transaction.getQuantity();
+        int currentQuantity = storageService.getQuantity(fruit);
+        if (currentQuantity >= quantity) {
+            storageService.remove(fruit, quantity);
+        } else {
+            throw new RuntimeException("Not enough " + fruit + " in storage");
+        }

```

Comment:
```suggestion
        if (currentQuantity < quantity) {
            throw new RuntimeException("Not enough " + fruit + " in storage");
        } 
        storageService.remove(fruit, quantity);
```
Avoid if-else blocks